internal/version/version.go

```diff
@@ -1,33 +1,32 @@
 package version
 
 import (
-	"fmt"
 	"runtime/debug"
 )
 
 var (
-	version = ""
+	version = "unknown"
 	sum     = ""
 )
 
-func init() {
+func Init() {
 	info, ok := debug.ReadBuildInfo()
 	if !ok || info.Main.Version == "" {
-		version = "unknown"
-	} else {
-		if version == "" {
-			version = info.Main.Version
-		}
-		if sum == "" {
-			sum = info.Main.Sum
-		}
+		return
 	}
```

Comment:
We need to keep these empty checks as these variables are pre-populated by `ldflags` in some package managers. The default value also needs to be empty for the same reason.

There have been quite a few accidental changes around this function. We should probably add some comments here to explain this too.

Reply:
Coming at it fresh it does seem a bit strange 😅 The reason for the change was a failing test that assumed it would be "unknown" but would return blank. What's the best approach to fix that? Set it in Init if it's not set in the buildinfo?

Reply:
This is because our tests treat Task as a package, not a CLI. This is a bit awkward to fix, but I think we could add a setting in the executor called `EnableVersionCheck` (Similar to the `DisableVersionCheck` you had previously. This setting would then need to be added to the CLI entrypoint and any relevant tests that need the version check functionality.